src/daemon/engine/svc/node/exec_cmd_service.cpp

```diff
@@ -0,0 +1,284 @@
+    ///
+    void operator()(Channel&& channel, const Spec::Request& request)
+    {
+        const auto fsm_id = next_fsm_id_++;
+        logger_->debug("New '{}' service channel (fsm={}).", Spec::svc_full_name(), fsm_id);
+
+        auto fsm           = std::make_shared<Fsm>(*this, fsm_id, std::move(channel));
+        id_to_fsm_[fsm_id] = fsm;
+
+        fsm->start(request);
+    }
+
+private:
+    // Defines private Finite State Machine (FSM) which tracks the progress of a single service request.
+    // There is one FSM per each service request channel.
+    //
+    // 1. On its `start` a set of Cyphal RPC clients is created (one per each node ID in the request),
+    //    and a command request is sent to each of them.
+    // 2. Then, the FSM waits for the RPC responses from the Cyphal nodes, collecting them.
+    // 3. Finally, FSM sends the accumulated result (when all responses are received or timed out)
+    //    and completes the channel.
+    //
+    class Fsm final
+    {
+    public:
+        using Id  = std::uint64_t;
+        using Ptr = std::shared_ptr<Fsm>;
+
+        Fsm(ExecCmdServiceImpl& service, const Id id, Channel&& channel)
+            : id_{id}
+            , channel_{std::move(channel)}
+            , service_{service}
+        {
+            logger().trace("ExecCmdSvc::Fsm (id={}).", id_);
+
+            channel_.subscribe([this](const auto& event_var) {
+                //
+                cetl::visit([this](const auto& event) { handleEvent(event); }, event_var);
+            });
+        }
+
+        ~Fsm()
+        {
+            logger().trace("ExecCmdSvc::~Fsm (id={}).", id_);
+        }
+
+        Fsm(const Fsm&)                = delete;
+        Fsm(Fsm&&) noexcept            = delete;
+        Fsm& operator=(const Fsm&)     = delete;
+        Fsm& operator=(Fsm&&) noexcept = delete;
+
+        void start(const Spec::Request& request)
+        {
+            logger().trace("ExecCmdSvc::Fsm::start (fsm_id={}).", id_);
+
+            // Immediately complete if there are no nodes to execute the command on.
+            //
+            if (request.node_ids.empty())
+            {
+                complete(0);
+                return;
+            }
+
+            // It's ok to have duplicates in the request -
+            // we just ignore duplicates, and work with unique ones.
+            const SetOfNodeIds unique_node_ids{request.node_ids.begin(), request.node_ids.end()};
+
+            const auto deadline = service_.context_.executor.now() + std::chrono::microseconds{request.timeout_us};
+            const CyphalExecCmdSvc::Request cy_request{request.payload.command, request.payload.parameter, &memory()};
+            for (const auto node_id : unique_node_ids)
+            {
+                if (const auto err = makeCyphalSvcCallFor(deadline, node_id, cy_request))
+                {
+                    complete(err);
```

Comment:
So, this isn't transactional. What's the ramification for partial success?

Reply:
Hmm... We had discussed with Pavel various paths of what could go wrong with such "bulk" requests. And we agreed on implementing "best efforts" strategy. For example, not responding nodes will not fail whole SDK request - the result map will just miss such nodes. But here I decided to break and fail whole request - mostly b/c such error is kinda unexpected one (the only reason why it could fail here is out of memory; maybe also out of range node id (like >=127 for CAN)). You are right that effect will be partial - all nodes before problematic one will receive the command. I think we can change a bit sdk contract so that `err` code will be attached for each node (and even for not responding, like ETIMEOUT).

Reply:
>I think we can change a bit sdk contract so that err code will be attached for each node (and even for not responding, like ETIMEOUT).

This is sensible